pkgs/development/hare-third-party/hare-png/default.nix

```
@@ -0,0 +1,40 @@
+{ lib, stdenv, hare, harec, hareThirdParty, fetchFromSourcehut }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hare-png";
+  version = "unstable-2023-09-09";
+
+  src = fetchFromSourcehut {
+    owner = "~sircmpwn";
+    repo = "hare-png";
+    rev = "9deb848ae691d784c5cc7949153cfa04c67f90b6";
+    hash = "sha256-Q7xylsLVd/sp57kv6WzC7QHGN1xOsm7YEsYCbY/zi1Q=";
+  };
+
+  nativeBuildInputs = [ hare ];
+  buildInputs = [ hareThirdParty.hare-compress ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HARECACHE="$NIX_BUILD_TOP/.harecache"
+    export BINOUT="$NIX_BUILD_TOP/.bin"
+
+    makeFlagsArray+=(
+      PREFIX="${builtins.placeholder "out"}"
+    )
+
+    runHook postConfigure
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~sircmpwn/hare-png/";
+    description = "PNG implementation for Hare";
+    license = with licenses; [ mpl20 ];
+    maintainers = with maintainers; [ starzation ];
+
+    inherit (harec.meta) platforms badPlatforms;
```

The same may be achieved through `hare`, removing the need of `harec` as a pattern.

```suggestion
    inherit (hare.meta) platforms badPlatforms;
```